pkg/reconciler/common/deployments_override.go

```diff
@@ -86,3 +87,9 @@ func replaceReplicas(override *v1alpha1.DeploymentOverride, deployment *appsv1.D
 		deployment.Spec.Replicas = &override.Replicas
 	}
 }
+
+func replaceNodeSelector(override *v1alpha1.DeploymentOverride, deployment *appsv1.Deployment) {
+	if len(override.NodeSelector) > 0 {
+		deployment.Spec.Template.Spec.NodeSelector = override.NodeSelector
```

Comment:
This feature is fine.
My question is whether to overwrite and add the nodeselector.
NodeSelector is a key-value map, the same to Labels and Annotations.
For Labels and Annotations, we add the override into the existing map, but for NodeSelector, we replace the existing with the override. Why not keep the existing keys, that is not in override for NodeSelector?

Reply:
I don't think it's useful to merge node selectors. IIRC they're in an `AND` combination, so adding more would likely result in no nodes matching anymore. Plus: We don't have preset node selectors at all.

Reply:
Yes I made this by design because of the reason Markus mentioned.

Reply:
Then I am ok with this PR.